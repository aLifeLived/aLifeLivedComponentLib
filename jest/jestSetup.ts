/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
