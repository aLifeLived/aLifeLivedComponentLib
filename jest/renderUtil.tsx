import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
