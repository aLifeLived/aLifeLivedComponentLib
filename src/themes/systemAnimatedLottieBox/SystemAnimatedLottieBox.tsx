import {
  backgroundColor,
  BackgroundColorProps,
  border,
  BorderProps,
  BoxProps,
  color,
  ColorProps,
  createRestyleComponent,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
} from '@shopify/restyle';
import React from 'react';
import AnimatedLottieView, {
  AnimatedLottieViewProps,
} from 'lottie-react-native';
import { Theme } from '../theme';

export type StyledSystemAnimatedLottieBox = AnimatedLottieViewProps &
  SpacingProps<Theme> &
  LayoutProps<Theme> &
  BorderProps<Theme> &
  BackgroundColorProps<Theme> &
  BoxProps<Theme> &
  ColorProps<Theme> & {
    children?: React.ReactNode;
  };

export const SystemAnimatedLottieBox = createRestyleComponent<
  StyledSystemAnimatedLottieBox,
  Theme
>([spacing, layout, border, backgroundColor, color], AnimatedLottieView);
