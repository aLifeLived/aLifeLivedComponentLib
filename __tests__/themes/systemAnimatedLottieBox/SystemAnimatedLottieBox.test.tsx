import React from 'react';
import { render } from '../../../jest/renderUtil';

// Component
import { SystemAnimatedLottieBox } from '../../../src/themes/systemAnimatedLottieBox/SystemAnimatedLottieBox';

const setup = () =>
  render(
    <SystemAnimatedLottieBox
      source={'./../../../src/assets/lottie/record-button.json'}
    />
  );

describe('themes >> systemAnimatedLottieBox >> SystemAnimatedLottieBox', () => {
  test('Should match snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });
});
