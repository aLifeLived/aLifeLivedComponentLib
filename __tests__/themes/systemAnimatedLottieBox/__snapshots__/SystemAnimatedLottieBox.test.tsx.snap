// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> systemAnimatedLottieBox >> SystemAnimatedLottieBox Should match snapshot 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <View
    autoPlay={false}
    cacheComposition={true}
    collapsable={false}
    enableMergePathsAndroidForKitKatAndAbove={false}
    forwardedRef={[Function]}
    loop={true}
    onAnimationFinish={[Function]}
    onLayout={[Function]}
    progress={0}
    resizeMode="contain"
    sourceName="./../../../src/assets/lottie/record-button.json"
    speed={1}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
